lib/std/fs/File.zig

```diff
@@ -1343,14 +1343,15 @@ pub const Reader = struct {
                 if (is_windows) {
                     // Unfortunately, `ReadFileScatter` cannot be used since it
                     // requires page alignment.
-                    assert(io_reader.seek == io_reader.end);
-                    io_reader.seek = 0;
-                    io_reader.end = 0;
+                    if (io_reader.seek == io_reader.end) {
+                        io_reader.seek = 0;
+                        io_reader.end = 0;
+                    }
                     const first = data[0];
-                    if (first.len >= io_reader.buffer.len) {
+                    if (first.len >= io_reader.buffer.len - io_reader.end) {
                         return readPositional(r, first);
                     } else {
-                        io_reader.end += try readPositional(r, io_reader.buffer);
+                        io_reader.end += try readPositional(r, io_reader.buffer[io_reader.end..]);
                         return 0;
                     }
                 }
```

Comment:
I *think* this entire branch (and the identical copy below) could be removed, since `posix.preadv/readv` is implemented as a single buffer `ReadFile` on Windows anyway.